### PR TITLE
API Rework

### LINKED ISSUE #684: Migrate API to django-modern-rpc
To get rid of our frankensteined fork of django-json-rpc and switch to something that wasn't last tested on Django 1.8, I would like to switch our API to [django-modern-rpc](https://github.com/alorence/django-modern-rpc). Hopefully this will also get rid of the random API errors we keep getting, or at least give us some proper error messages for them.

This will require me to ensure that we do not break our current API specification, method names, etc.

### PR DESCRIPTION
**Pull request by Kurocon**
Please add the following information to your pull request:

**Please describe what your PR is fixing**
Re-write the API so it uses a more modern JSONRPC library, so we can lose one of our 'weird' dependencies.
Also moves the documentation previously hosted [here](https://github.com/Inter-Actief/api-docs/) to the codebase, so it can be auto generated.

**Concretely, which issues does your PR solve? (Please reference them by typing `Fixes/References Inter-Actief/amelie#<issue_id>`)**
Fixes #684

Hopefully also fixes these other API issues, but not sure:
References #601 
References #584

**Does your PR change how we process personal data, impact our [privacy document](https://www.inter-actief.utwente.nl/privacy/), or modify (one of) our data export(s)?**
no

**Does your PR include any django migrations?**
no

**Does your PR include the proper translations (did you add translations for new/modified strings)?**
no, my PR does not include translations

**Does your PR include CSS changes (and did you run the `compile_css.sh` script in the `scripts` directory to regenerate the `compiled.css` file)?**
no, my PR does not include CSS changes

**Does your PR need external actions by for example the System Administrators? (Think about new pip packages, new (local) settings, a new regular task or cronjob, new management commands, etc.)?**
no

**Did you properly test your PR before submitting it?**
kind of

~~**TODO:** Needs proper testing using the actual IA app.
However, this is a bit difficult because we don't have a working build where we can modify the backend API that is used. I will try some sysadmin magic to see if I can make only _my_ app redirect to this API running on staging or a beta environment, to see if it works properly.~~

~~If I can't get this working, the API tests _should_ also be enough to ensure that the app also keeps working...
With postman the responses seem to be correct.~~

Tested with the actual app and everything works after some small fixes for weird behaviour that the app expects from the API.

### CODE REVIEW

**Comment by Kurocon** [comment]
The API tests seem to be very strict to the output of the old library, but the source code of the App itself seems to be more accepting. Next time I'll fix up the tests and do some more testing with the app and the narrowcasting/cookie corner screens that also use the API.

**Comment by Kurocon** [comment]
Ready for review @WKobes @bramvandartel

File: amelie/tools/tests.py [hunk 1]
```diff
@@ -169,44 +169,36 @@ def send_and_compare_request(self, method, params, token, expected_result):
...
-            },
-            'result': None,
-        }
+        json_content = json.loads(content)
 
-        self.assertJSONEqual(content, expected_data, 'JSON RPC result')
+        # Compare jsonrpc version
+        self.assertEqual(json_content.get("jsonrpc", None), "2.0")
+        # Compare id
+        self.assertEqual(json_content.get("jsonrpc", None), "2.0")
```
Comment (WKobes) [inline amelie/tools/tests.py:197]: duplicate test?

**Review by WKobes — COMMENTED** [review]
Nice work! Just added one comment on a test

**Review by WKobes — APPROVED** [review]

File: amelie/api/education.py [hunk 2]
```diff
@@ -1,21 +1,75 @@
...
+        - dateTime: The submission date and time of this complaint (RFC3339).
+        - type: The type of this complaint.
+        - summary: The summary of this complaint.
+        - remark: The content of the complaint.
+        - supporters: The amount of supporters for this complaint (includes the original reporter).
+        - hasSupported: true if the authenticated person has supported this complaint, otherwise false.
+        - course: The course concerning this complaint, can be null.
+        - subject: The subject concerning the course of this complaint, can be null.
+        - year: The first year of the academic year in which the course was given, can be null.
+        - period: The period of the academic year in which the course was given, can be null.
```
Comment (Kurocon) [inline amelie/api/education.py:45]: isCompleted mist in docstring

File: amelie/api/person.py [hunk 3]
```diff
@@ -1,25 +1,103 @@
...
+        - candidateBoard: true if this person is a candidate board member, false otherwise
+        - member: true if this person is a member, false otherwise
+        - hasCommittees: true if this member has joined any committees, false otherwise
+        - hasActivityDirectDebit: true if this person has signed a debit contract, false otherwise
+        - username: The username of this person, can be empty or null
+        - studentId: The student ID of this person, can be null
+        - employeeId: The employee ID of this person, can be null
+        - email: The email address of this person
+        - languagePreference: The preferred language for this user, either "nl" or "en"
+        - imageUrl: The profile image URL for the authenticated user
```
Comment (Kurocon) [inline amelie/api/person.py:73]: Can be null

File: amelie/api/person.py [hunk 4]
```diff
@@ -83,10 +194,52 @@ def get_person_committees(request, authentication=None):
...
+
+    **Parameters:** _(none)_
+
+    **Return:**
+      `Union[Dict, None]`: A dictionary with the following fields, or null if the person has no active membership:
+
+        - type: The type of membership
+        - year: The year of the active membership
+        - hasEnded: Value indicating whether the membership has ended, either true or false
+        - payment: Details regarding the payment of this membership
```
Comment (Kurocon) [inline amelie/api/person.py:215]: Payment can be null

File: amelie/api/person.py [hunk 5]
```diff
@@ -54,15 +132,48 @@ def get_person_details(request, authentication=None):
...
+
+    **Parameters:** _(none)_
+
+    **Return:**
+      `Union[List[Dict], None]`: An array of dictionaries containing the committee information,
+                                 or null if the current user has no associated person.
+
+      Each returned element in the list has the following fields:
+
+        - position: The position of this person in this committee
```
Comment (Kurocon) [inline amelie/api/person.py:158]: Docs not up to date with actual returned structure

File: amelie/api/videos.py [hunk 6]
```diff
@@ -18,13 +21,72 @@ def get_basic_result(video):
...
-    videos = YouTubeVideo.objects.filter_public(not authenticated).order_by('-date_published')[offset:length + offset]
+      Each returned element in the list has the following fields:
+
+        - id: The identifier for this video
+        - type: The type of video, one of "yt" (YouTube) or "ia" (Streaming.IA)
+        - title: The title of this video
+        - thumbnail: The URL to the thumbnail of this video
+        - featured: Boolean, true if the video is currently being featured, false if not.
+        - date: The date this video was published (RFC3339)
+        - publisher: A dictionary containing information about the publishing committee, including the following fields:
```
Comment (Kurocon) [inline amelie/api/videos.py:52]: Can be null

File: amelie/api/videos.py [hunk 7]
```diff
@@ -34,13 +96,81 @@ def get_video_stream(request, offset, length, authentication=None):
...
+
+      Each returned element in the list has the following fields:
+
+        - id: The identifier for this video
+        - type: The type of video, one of "yt" (YouTube) or "ia" (Streaming.IA)
+        - title: The title of this video
+        - thumbnail: The URL to the thumbnail of this video
+        - featured: Boolean, true if the video is currently being featured, false if not.
+        - date: The date this video was published (RFC3339)
+        - publisher: A dictionary containing information about the publishing committee, including the following fields:
```
Comment (Kurocon) [inline amelie/api/videos.py:125]: Can be null
